pkg/providers/instancetype/types.go

```diff
@@ -137,6 +141,45 @@ func computeRequirements(ctx context.Context, info *ec2.InstanceTypeInfo, offeri
 	return requirements
 }
 
+func getOs(info *ec2.InstanceTypeInfo, family amifamily.AMIFamily) string {
```

Comment:
Do we need to take a dependency on the AMIFamily here? Can we just special-case the instance types that support windows and add a `windows` value to their `os` requirement?

Reply:
A two-values requirement prevents Karpenter from creating the node labels.
So the test launchtemplate_test.suite_test.go `should apply labels to the node` fails at 
`Expect(node.Labels).To(HaveKey(v1.LabelOSStable)`

Can we remove this "expect" since kubelet should label node correct at runtime? We don't need any tests for these Well-Known labels from my understanding.

Reply:
> two-values requirement prevents Karpenter from creating the node labels

I think these labels should constrain once the instance is launched. If it doesn't do that right now, then we should constrain and add the labels at launch-time for OS (possibly from the AMI), even if there are two values in the requirement.

I don't think constraining the requirements on the instance type to get labels to propagate on the node is the right separation of concerns though.

Reply:
Done. AMIFamily is not required.